### PR TITLE
yy-choose

### PR DESCRIPTION
**Pull request by yuanyi-350**

### CODE REVIEW

**Comment by yuanyi-350** [comment]
I suggest placing this section after the Teaching Structure section. P.S. The structure will help in understanding type transformation

**Comment by timechess** [comment]
Put your exercises in another file like `choose.lean`

**Comment by yuanyi-350** [comment]
Ok, I have already revised it.

File: TacticsInLean/Choose.lean [hunk 1]
```diff
@@ -0,0 +1,269 @@
+import Mathlib.Tactic
+
+
+open Classical
+suppress_compilation
+
+
+section Introduction1
```
Comment (timechess) [inline TacticsInLean/Choose.lean:8]: I think there should be a more gentle example(even without sorry just to show the usage) before this section

File: TacticsInLean/Choose.lean [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+example (X : Type) (P : X → ℝ → Prop)(h : ∀ ε > 0, ∃ x, P x ε) : ∃ u : ℕ → X, ∀ n, P (u n) (1/(n+1)) := by
+  --use the `choose` tactic to replace these "sorry"s
+  let g : (ε : ℝ) → ε > 0 → X := sorry
+  have hg : ∀ (ε : ℝ) (a : ε > 0), P (g ε a) ε := sorry
```
Comment (timechess) [inline TacticsInLean/Choose.lean:12]: I don't think do it separately will be a good idea. The example provided in `#help tactic` doesn't show how to do this.  You should provide an example or change it.

File: TacticsInLean/Choose.lean [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+
+open Set
+
+theorem myexists_eq_graphOn_image_fst{α : Type*} {β : Type*} [Nonempty β] {s : Set (α × β)}
+(h : Set.InjOn Prod.fst s) : ∃ (f : α → β), s = Set.graphOn f (Prod.fst '' s):= by
+  have : ∀ x ∈ Prod.fst '' s, ∃ y, (x, y) ∈ s := forall_mem_image.2 fun (x, y) h ↦ ⟨y, h⟩
+
+  --use `choose` tactic to replace these "sorry"s
+  let f : α → β := sorry
+  have hf : ∀ x ∈ Prod.fst '' s, (x, f x) ∈ s := sorry
```
Comment (timechess) [inline TacticsInLean/Choose.lean:128]: Same as above

File: TacticsInLean/Choose.lean [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+
+end Exercise2
+
+
+
+section Exercise3
+
+open Pointwise
+
+def Hom_top_product_of_normal_of_disjoint (H K : Set ℝ) : (H * K) → (H × K) := by
```
Comment (timechess) [inline TacticsInLean/Choose.lean:158]: You should provide natural language description about this definition.

**Review by timechess — CHANGES_REQUESTED** [review]

**Review by timechess — APPROVED** [review]

**Pull request merged**